tests/Meziantou.Analyzer.Test/Rules/FileNameMustMatchTypeNameAnalyzerTests.cs

```diff
@@ -247,6 +247,19 @@ struct Bar {}
               .ValidateAsync();
     }
 
+    [Fact]
+    public async Task MatchOnlyFirstType_TypeWithNamespaceDeclaration()
```

Comment:
You need to surround the test with:

```c#
#if CSHARP10_OR_GREATER
```

Locally, you use the latest version of Roslyn, but the CI runs against multiple versions of Roslyn, including 3.8 (needed for older versions of VS and .NET SDK) which does not support some new features.

Reply:
Or I can you block scoped namespace. I guess it's better for quality control.
Thx for info, will update PR today.

Reply:
Or you can do both :) (there are never enough tests for a Roslyn analyzer)